pkg/timeline/service/push.ts

```diff
@@ -0,0 +1,55 @@
+import { Result } from '@mikuroxina/mini-fn';
+
+import type { AccountModule } from '../../intermodule/account.js';
+import type { Note } from '../../notes/model/note.js';
+import type { TimelineNotesCacheRepository } from '../model/repository.js';
+import type { NoteVisibilityService } from './noteVisibility.js';
+
+export class PushTimelineService {
+  constructor(
+    private readonly accountModule: AccountModule,
+    private readonly noteVisibility: NoteVisibilityService,
+    private readonly timelineNotesCacheRepository: TimelineNotesCacheRepository,
+  ) {}
+
+  /**
+   * @description Push note to home timeline
+   * @param note to be pushed
+   * */
+  async handle(note: Note): Promise<Result.Result<Error, void>> {
+    const followers = await this.accountModule.fetchFollowers(
+      note.getAuthorID(),
+    );
+    if (Result.isErr(followers)) {
+      return followers;
+    }
+    const unwrappedFollowers = Result.unwrap(followers);
+
+    /*
+    PUBLIC, HOME, FOLLOWER: OK
+    DIRECT: reject (direct note is not pushed to home timeline)
+     */
+    const isNeedReject =
+      !(await this.noteVisibility.homeTimelineVisibilityCheck({
+        accountID: note.getAuthorID(),
+        note,
+      }));
+    if (isNeedReject) {
+      return Result.err(new Error('Note is not visible'));
+    }
```

Comment:
```suggestion
    const visible =
      await this.noteVisibility.homeTimelineVisibilityCheck({
        accountID: note.getAuthorID(),
        note,
      });
    if (!visible) {
      return Result.err(new Error('Note is invisible'));
    }
```
否定形の変数名は読みにくいので, 肯定形で保持して条件を否定にしたほうがよいと思います.